lib/Listener/CalendarReferenceListener.php

```diff
@@ -0,0 +1,130 @@
+use OCP\Collaboration\Reference\RenderReferenceEvent;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\Files\IAppData;
+use OCP\IConfig;
+
+
+
+use OCP\Util;
+
+/**
+ * @template-implements IEventListener<Event|RenderReferenceEvent>
+ */
+class CalendarReferenceListener implements IEventListener {
+
+	/** @var IInitialState */
+	private $initialStateService;
+
+	/** @var IAppManager */
+	private $appManager;
+
+	/** @var IConfig */
+	private $config;
+	
+	/** @var CompareVersion */
+	private $compareVersion;
+
+	private IAppData $appData;
+
+	public function __construct(
+		IInitialState $initialStateService,
+		IAppManager $appManager,
+		IConfig $config,
+		IAppData $appData,
+		CompareVersion $compareVersion,
+	) {
+		$this->config = $config;
+		$this->initialStateService = $initialStateService;
+		$this->appManager = $appManager;
+		$this->appData = $appData;
+		$this->compareVersion = $compareVersion;
+
+	}
+
+	public function handle(Event $event): void {
+		if (!$event instanceof RenderReferenceEvent) {
+			return;
+		}
+		$defaultEventLimit = $this->config->getAppValue('calendar', 'eventLimit', 'yes');
+		$defaultInitialView = $this->config->getAppValue('calendar', 'currentView', 'dayGridMonth');
+		$defaultShowWeekends = $this->config->getAppValue('calendar', 'showWeekends', 'yes');
+		$defaultWeekNumbers = $this->config->getAppValue('calendar', 'showWeekNr', 'no');
+		$defaultSkipPopover = $this->config->getAppValue('calendar', 'skipPopover', 'no');
+		$defaultTimezone = $this->config->getAppValue('calendar', 'timezone', 'automatic');
+		$defaultSlotDuration = $this->config->getAppValue('calendar', 'slotDuration', '00:30:00');
+		$defaultDefaultReminder = $this->config->getAppValue('calendar', 'defaultReminder', 'none');
+
+		$appVersion = $this->config->getAppValue('calendar', 'installed_version', '');
+		$forceEventAlarmType = $this->config->getAppValue('calendar', 'forceEventAlarmType', '');
+		if (!in_array($forceEventAlarmType, ['DISPLAY', 'EMAIL'], true)) {
+			$forceEventAlarmType = false;
+		}
+		$showResources = $this->config->getAppValue('calendar', 'showResources', 'yes') === 'yes';
+		$publicCalendars = $this->config->getAppValue('calendar', 'publicCalendars', '');
+
+		$talkApiVersion = version_compare($this->appManager->getAppVersion('spreed'), '12.0.0', '>=') ? 'v4' : 'v1';
+		$tasksEnabled = $this->appManager->isEnabledForUser('tasks');
+
+		$circleVersion = $this->appManager->getAppVersion('circles');
+		$isCirclesEnabled = $this->appManager->isEnabledForUser('circles') === true;
+		// if circles is not installed, we use 0.0.0
+		$isCircleVersionCompatible = $this->compareVersion->isCompatible($circleVersion ? $circleVersion : '0.0.0', '22');
+
+		$this->initialStateService->provideInitialState('app_version', $appVersion);
+		$this->initialStateService->provideInitialState('event_limit', $defaultEventLimit);
+		$this->initialStateService->provideInitialState('first_run', true);
```

Comment:
```suggestion
		$this->initialStateService->provideInitialState('first_run', false);
```
Fun fact, it seems this config isn't even used anymore.